File: Domain/Model/DnevnikPromena.cs
using System;

namespace Domain.Model
{
    public class DnevnikPromena : IAccessImportSourceLineage
    {
        public int Id { get; set; }
        /// <summary>Movement type. Use <see cref="TipPromeneConstants"/> constants for all comparisons.</summary>
        public string TipPromene { get; set; } = string.Empty;
        public DateTime Datum { get; set; }
        public decimal Iznos { get; set; }
        public string? BrojRacuna { get; set; }
        public int? DobavljacId { get; set; }

        public int? ArtikalId { get; set; }
        public decimal? StaraProdajnaCena { get; set; }
        public decimal? NovaProdajnaCena { get; set; }

        public int? Kolicina { get; set; }      // quantity involved in this change (e.g. pieces sold/received)
        public int? IDObjekat { get; set; }     // store/objekat for movement tracking and per-store analytics
        public int? RedniBroj { get; set; }     // original sequence/line number from source system (audit trail)
        public string? Komentar { get; set; }
        public string? KorisnikIme { get; set; }
        public string DataOrigin { get; set; } = "existing";
        public string? SourceTableKey { get; set; }
        public long? SourceRowId { get; set; }
        public DateTime? SourceUpdatedAtUtc { get; set; }
        public string? SourceHash { get; set; }
        public long? SourceBatchId { get; set; }
    }
}

File: Klijent/clientapp/src/types/dnevnikPromena.ts
export interface DnevnikPromenaItem {
  id: number;
  tipPromene: string;
  datum: string;
  iznos: number;
  brojRacuna?: string | null;
  artikalId?: number | null;
  artikalNaziv?: string | null;
  dobavljacId?: number | null;
  dobavljacNaziv?: string | null;
  staraProdajnaCena?: number | null;
  novaProdajnaCena?: number | null;
  komentar?: string | null;
  korisnikIme?: string | null;
  dataOrigin?: string | null;
}

export interface DnevnikPromenaResponse {
  items: DnevnikPromenaItem[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  sortBy: string;
  sortDir: string;
}

export interface DnevnikPromenaDetail {
  id: number;
  tipPromene: string;
  datum: string;
  artikalId?: number | null;
  nazivArtikla?: string | null;
  kolicina?: number | null;
  staraCena?: number | null;
  novaCena?: number | null;
  iznos: number;
  brojRacuna?: string | null;
  korisnikIme?: string | null;
  komentar?: string | null;
  dataOrigin?: string | null;
  sourceId: number;
}
